### PR TITLE
[MPS] Add MPSHooks interface to enable accessing MPS functions globally

### PR DESCRIPTION
**Pull request by razarmehr**
This PR is a prerequisite to the upcoming MPSGenerator changes required for Random Ops.

Add `MPSHooksInterface.cpp` to `aten_cpu_source_non_codegen_list`

Co-authored-by: Nikita Shulga <nikita.shulga@gmail.com>

cc @kulinseth @albanD @malfet @DenisVieriu97 @abhudev

### CODE REVIEW

**Review by kulinseth — APPROVED** [review]

File: aten/src/ATen/detail/MPSHooksInterface.cpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+const MPSHooksInterface& getMPSHooks() {
+  static std::unique_ptr<MPSHooksInterface> mps_hooks;
+#if !defined C10_MOBILE
+  static c10::once_flag once;
+  c10::call_once(once, [] {
+    mps_hooks = MPSHooksRegistry()->Create("MPSHooks", MPSHooksArgs{});
+    if (!mps_hooks) {
+      mps_hooks =
+          // NOLINTNEXTLINE(modernize-make-unique)
+          std::unique_ptr<MPSHooksInterface>(new MPSHooksInterface());
```
Comment (malfet) [inline aten/src/ATen/detail/MPSHooksInterface.cpp:19]:
```suggestion
      mps_hooks = std::make_unique<MPSHooksInterface>();
```

File: aten/src/ATen/detail/MPSHooksInterface.cpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    }
+  });
+#else
+  if (mps_hooks == nullptr) {
+    mps_hooks =
+        // NOLINTNEXTLINE(modernize-make-unique)
+        std::unique_ptr<MPSHooksInterface>(new MPSHooksInterface());
```
Comment (malfet) [inline aten/src/ATen/detail/MPSHooksInterface.cpp:26]: Indeed, modernize your C++ :)
```suggestion
    mps_hooks = std::make_unique<MPSHooksInterface>();
```

**Review by malfet — APPROVED** [review]
LGTM, but please address nits

**Comment by malfet** [comment]
Hmm, lite interpreter failures seem relevant:
```
 Undefined symbols for architecture x86_64:
   "at::MPSHooksRegistry()", referenced from:
       ___cxx_global_var_init in MPSHooks.cpp.o
   "at::detail::getMPSHooks()", referenced from:
       at::Context::hasMPS() in EmptyTensor.cpp.o
       at::Context::hasMPS() in MPSGuardImpl.mm.o
 ld: symbol(s) not found for architecture x86_64
```

**Comment by malfet** [comment]
This should probably do the trick: https://github.com/pytorch/pytorch/pull/91104/commits/dcc66e8187ba866e49f43f996370dfce19278995

**Pull request closed without merging**